### PR TITLE
[IMP][FIX] purchase_stock_price_unit_sync: handle kit cases

### PR DESCRIPTION
**Pull request by chienandalu**
If we receive a kit, we shouldn't write the kit purchase price on the
components price units.

TT32040

ping @sergio-teruel

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Why `_get_stock_move_price_unit` is not returning the product standard price?

**Comment by chienandalu** [comment]
It gets the kit cost, not the components ones

**Comment by pedrobaeza** [comment]
But that's because the standard? Shouldn't this be changed in Odoo itself?

**Comment by chienandalu** [comment]
Well, as far as we could find out, the exploded component moves default to the current standard price in the moment of the validation as the price_unit of the purchase line isn't propagated (and that it's correct, or otherwise we'll be having wrong valuations, which is the point of the fix).

I'm going to try and get the immediate previous historic price so the cost is more reliable.

We can't do much better I'm afraid.

**Review by pedrobaeza — APPROVED** [review]

File: purchase_stock_price_unit_sync/models/purchase_order.py [hunk 1]
```diff
@@ -16,6 +16,18 @@ def write(self, vals):
...
         for line in self.filtered(lambda l: l.state in ['purchase', 'done']):
+            # When we receive a kit, we can only default to its registered cost.
+            # We are avoiding a hard dependency here though.
+            if (
+                hasattr(line.product_id, "bom_ids")
+                and line.product_id._is_phantom_bom()
+            ):
+                for move in line.move_ids:
+                    previous_instant = move._previous_instant_date()
+                    move.price_unit = move.product_id.get_history_price(
```
Comment (sergio-teruel) [inline purchase_stock_price_unit_sync/models/purchase_order.py]: Why do we update the price with the previous price??

**Review by sergio-teruel — APPROVED** [review]

**Review by sergio-teruel — CHANGES_REQUESTED** [review]
Sorry

File: purchase_stock_price_unit_sync/models/purchase_order.py (hunk 1 above)
Comment (chienandalu) [reply purchase_stock_price_unit_sync/models/purchase_order.py]: Ok, as we debated internally it's better to do nothing as that's what core does anyway and we don't want resync the whole move chain

**Review by sergio-teruel — APPROVED** [review]

**Pull request merged**